contents/docs/data/events.mdx

```diff
@@ -114,6 +114,64 @@ Most users of PostHog will want to combine their back-end data, such as user inf
 -   JS [snippet](/docs/integrate)
 -   Client- and server-side [libraries](/docs/integrate)
 
+## Events with people processing
+
+By default all events create or update an associated [person profile](/docs/data/persons). This allows for the following features:
+
+- Filtering on person properties
+- Merging identities across distinct_ids (ie merging anonymous users with your logged in user)
+- Cohort creation
+- Feature flag targeting
+- Tracking initial UTM tags and referrers over time
+
+However, there are many usecases that don't require this person processing. Because events that don't do person processing are cheaper for us to ingest, the pricing for these events is lower.
+
+Some examples of usecases that often don't require user profiles
+- Tracking a marketing website
+- B2C apps where users don't sign up or log in
+- High volume API server events
+
+## PostHog-js
+
+You can control whether events from PostHog-js have person processing enabled with the `person_processing` config setting. There are three options:
+
+- `person_processing: 'always'` _(default)_ - we will process persons data for all events
+- `person_processing: 'identified_only'` - we will only process persons when you call `posthog.identify([distinct_id])`. Anonymous users won't get person profiles.
+- `person_processing: 'never'` - we won't process persons for any event. This means that anonymous users will not be merged once they sign up or login, so you lose the ability to create funnels that track users from anonyomous to identified.
```

Comment:
I don't think this makes sense in the context of posthog-js. All posthog-js events have $set and $set_once properties, so every event would be charged a lot. I think the concept of anonymous_users = cheap, identified users = expensive makes sense here.

Reply:
Hm I don't think we should allow/encourage users to send person properties, but not process all events for that person. Otherwise with PoE you wouldn't be able to query those events using person properties right?

Reply:
So the thing is within ingestion we have 
- geoIP app enabled for everyone 
- the default extract utm etc initial properties
and we need to know to ignore that, which we can do if we have the boolean flag on every event (if it's not there we assume we do process persons, like currently)

Reply:
> Hm I don't think we should allow/encourage users to send person properties, but not process all events for that person. Otherwise with PoE you wouldn't be able to query those events using person properties right?

The way PoE works for each property (key): 
1. `event.properties.$set[key]` if set, else
2. `person?.properties[key]` if set (note that if person doesn't exist we can treat it as property doesn't exist), else
3. `event.properties.$set_once[key]` if set

This already works this way, because we want to allow each event to override PoE, because of the way we don't guarantee ordered processing across different distinct_ids. This is also what web analytics can use for sessions table (and is using afaik).